docs/developer-docs/functionality/http_requests/http_requests-how-it-works.md

```diff
@@ -137,16 +137,26 @@ We recommend to go with the first approach whenever possible as it has multiple
 * The transformation function is faster to compute, that is, the function (query) is executing with fewer CPU cycles. Note that queries are likely to be charged for in the future.
 * The transformation function is easier to implement, e.g., through a simple or few Json operations on the response body.
 
-### Pricing
+### Errors and Debugging
 
-FIX: update with the new pricing model which is likely substantially cheaper
+There are a number of error cases that can happen when using this feature. The most important ones are listed next.
+* *SysFatal - Url needs to specify https scheme:* The feature currently only allows for HTTPS connections and HTTP leads to an error.
+* *SysFatal - Timeout expired:* Requests are timed out if not fulfilled within the timeout period. One important instance when this happens is when there are not sufficiently many equal responses to achieve consensus.
+* *SysTransient - Failed to connect: error trying to connect: tcp connect error: Connection refused (os error 111):* This error indicates that a TCP connection could not have been established.
+* *CanisterReject - http_request request sent with 0 cycles, but ... cycles are required:* At least the required amount of cycles need to be sent with the request in order to it getting fulfilled by the subnet.
+* *CanisterReject - max_response_bytes expected to be in the range [0..2097152], got ...:* This error indicates that the network response received from the server was too large. This happens if the response size is underestimated and the `max_response_bytes` value set too low.
+* *SysFatal - Transformed http response exceeds limit: 2045952:* This error indicates that the limit for the transformed response size was hit. This is currently a hard limit of the HTTPS outcalls functionality.
 
-As (almost) every feature of the IC, the canister HTTP request feature is charged for when being used. The current pricing is defined to charge a base fee of $400$M cycles for an HTTP request in addition to $100$K cycles per request byte and per `max_response_bytes` byte. Because of the fixed cost and the overhead of HTTP requests including headers, it is advantageous to make fewer larger requests to retrieve the same amount of information than with more smaller requests, if this is feasible from the application perspective. The cycles provided with the call must be sufficient for covering the cost, remaining cycles are returned to the caller.
-The current pricing is defined to be very conservative (expensive) and prices are likely to change to the cheaper with the introduction of an update of the pricing model.
+Developers new to the feature are likely to run into certain problems in the beginning, materializing in one of the errors. We list the issues we think are the most prominent when starting with this feature.
```

Comment:
```suggestion
Developers new to the feature are likely to run into certain problems in the beginning, materializing in one of the above errors. We list the issues we think are the most prominent when starting with this feature.
```

Reply:
There was another suggestion that was accepted.